### PR TITLE
feature(data): add deployments, daemonsets to cluster data

### PR DESCRIPTION
**Pull request by jackfrancis**
`installedDeisData.Get()` now employs the `Deployments()`, `DaemonSets()` k8s interfaces in addition to `ReplicationControllers()` in order to present a complete set of workflow components when gathering cluster data.

Includes basic test coverage, with the caveat that because our current interfaces act as a union between the `api` and `extensions` resources, we'll probably need to build out our own fake client library in order to fully test.

### CODE REVIEW

**Comment by mboersma** [comment]
> we'll probably need to build out our own fake client library in order to fully test

If that's out of scope for the next Workflow release, let's make sure to create an issue to track it.

**Comment by jackfrancis** [comment]
@mboersma good idea: https://github.com/deis/workflow-manager/issues/96

File: data/installed_data.go [hunk 1]
```diff
@@ -25,19 +33,62 @@ func NewInstalledDeisData(ri *k8s.ResourceInterfaceNamespaced) InstalledData {
...
+			Version: &models.Version{
+				Version: deployment.Annotations[versionAnnotation],
+				Data: &models.VersionData{
+					Image: &deployment.Spec.Template.Spec.Containers[0].Image,
+				},
+			},
+		}
+		cluster.Components = append(cluster.Components, &component)
+	}
+	daemonSets, err := k8s.GetDaemonSets(g.k8sResources.DaemonSets())
```
Comment (arschles) [inline data/installed_data.go]: is `k8s.GetDaemonSets()` just a wrapper around `g.k8sResources.DaemonSets()`?

Comment (arschles) [inline data/installed_data.go]: nevermind, I just read below. the function looks trivial but it replaces >1 loc, so I'm ok with leaving it

File: data/installed_data.go [hunk 2]
```diff
@@ -25,19 +33,62 @@ func NewInstalledDeisData(ri *k8s.ResourceInterfaceNamespaced) InstalledData {
...
+		return nil, err
+	}
+	for _, daemonSet := range daemonSets {
+		component := models.ComponentVersion{
+			Component: &models.Component{
+				Name: daemonSet.Name,
+				Type: &daemonSetType,
+			},
+			Version: &models.Version{
+				Version: daemonSet.Annotations[versionAnnotation],
```
Comment (arschles) [inline data/installed_data.go]: seems like the version annotation will always exist, but this map lookup should still be checked. seems like the correct thing is to log a warning if the key is missing, but proceed anyway

File: data/installed_data.go (hunk 1 above)
Comment (jackfrancis) [inline data/installed_data.go]: It's a fine question. The higher level answer is that `k8s.GetDaemonSets()` is an exportable function that takes a *namespaced* Daemon Set interface (which is what `g.k8sResources.DaemonSets()` returns.

File: data/installed_data.go [hunk 3]
```diff
@@ -25,19 +33,62 @@ func NewInstalledDeisData(ri *k8s.ResourceInterfaceNamespaced) InstalledData {
...
+	for _, rc := range replicationControllers {
+		component := models.ComponentVersion{
+			Component: &models.Component{
+				Name: rc.Name,
+				Type: &rcType,
+			},
+			Version: &models.Version{
+				Version: rc.Annotations[versionAnnotation],
+				Data: &models.VersionData{
+					Image: &rc.Spec.Template.Spec.Containers[0].Image,
```
Comment (arschles) [inline data/installed_data.go]: same here regarding a check for # of containers - I know there should always be 1

File: data/installed_data.go (hunk 2 above)
Comment (jackfrancis) [inline data/installed_data.go]: thanks! https://github.com/deis/workflow-manager/issues/97

File: data/installed_data_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+const namespace = "deis"
+
+func TestInstalledDeisData(t *testing.T) {
+	client := getK8sClient(t)
+	k := k8s.NewResourceInterfaceNamespaced(client, namespace)
+	installedData := installedDeisData{
+		k8sResources: k,
+	}
+	_, _ = installedData.Get()
+	//TODO: we need to create a fake client interface that is a union of api+extension fake clients
```
Comment (arschles) [inline data/installed_data_test.go]: does this need doing before this patch is merged? if not, can you ensure there's an issue in for this TODO

File: data/installed_data.go (hunk 3 above)
Comment (jackfrancis) [inline data/installed_data.go]: thanks! https://github.com/deis/workflow-manager/issues/97

File: data/installed_data_test.go (hunk 4 above)
Comment (jackfrancis) [inline data/installed_data_test.go]: yep: https://github.com/deis/workflow-manager/issues/96

**Comment by jackfrancis** [comment]
fixes #90

**Pull request merged**